app/Nova/Actions/AddPayment.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Nova\Actions;
+
+use App\Payment;
+use Illuminate\Bus\Queueable;
+use Laravel\Nova\Fields\Select;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Fields\Currency;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Nova\Fields\ActionFields;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+
+class AddPayment extends Action
+{
+    use InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Perform the action on the given models.
+     *
+     * @param  \Laravel\Nova\Fields\ActionFields  $fields
+     * @param  \Illuminate\Support\Collection  $models
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        if (count($models) > 1) {
+            return Action::danger(
+                'Payments cannot be entered for more than one transaction at a time. Contact the developers if you think this is unreasonable.'
+            );
+        }
+        // shouldn't happen but might if someone is abusing the API
+        if (Auth::user()->cant('create-payments-'.$fields->method)) {
+            $this->markAsFailed($models->first(), null);
+
+            return Action::danger(
+                'You do not have permission to accept that payment method. Please contact a developer.'
+            );
+        }
+
+        if ($fields->method === 'square' || $fields->method === 'swipe') {
+            if ($fields->amount !== ($models->first()->package()->get()->first()->cost) + 3) {
+                $this->markAsFailed($models->first(), null);
+
+                return Action::danger(
+                    'Missing expected transaction fee - total should be '.(($models->first()->package()->get()->first()->cost) + 3).', '.$fields->amount.' entered.'
```

Comment:
I'm somewhat opposed to hardcoding the transaction fee.

Reply:
Do you have an alternative proposal?

Reply:
You could put something in .env though that still seems kinda wrong. Approving for now with the requirement that it be revisited later if we ever think of something better.

Reply:
Follow up - `5.00` != `5`, per Slack.